### PR TITLE
Add support for Derby/Sybase/SQLite/DB2/OceanBase jdbc url format in URLParser

### PR DESCRIPTION
**Pull request by CzyerChen**
### https://github.com/apache/skywalking/issues/12207
- [x] Tests(including UT, IT, E2E) are added to verify the new feature.
- [x] If this pull request closes/resolves/fixes an existing issue, replace the issue number. Closes https://github.com/apache/skywalking/issues/12207
- [x] Update the [`CHANGES` log](https://github.com/apache/skywalking-java/blob/main/CHANGES.md).

Support URLs

|DataBase|UrlPattern|URLs|
|--|--|--|
|Derby|  1、`jdbc:derby:[subprotocol:][databaseName][;attribute=value]*` <br/> 2、`jdbc:derby://server[:port]/databaseName[;attribute=value]*`|jdbc:derby:directory:mydb<br/> jdbc:derby:memory:mydb;create=true<br/>jdbc:derby:memory:C:\test\mydb<br/>jdbc:derby:memory:/home/test/mydb<br/>jdbc:derby:classpath:/test/mydb<br/>jdbc:derby:jar:(C:/dbs.jar)test/mydb<br/>jdbc:derby:mydb<br/>jdbc:derby:/test/mydb;create=true;create=true<br/>jdbc:derby:test/mydb;create=true<br/>jdbc:derby:A:/test/mydb;create=true<br/>jdbc:derby:test/mydb;create=true<br/>jdbc:derby://localhost:1527/mydb;create=true;user=root;password=pass<br/>jdbc:derby://localhost:1527/mydb<br/>jdbc:derby://localhost/mydb;create=true<br/>jdbc:derby://localhost:1527/c:\temp\mydb<br/>jdbc:derby://localhost:1527//opt/test/mydb;create=true<br/>jdbc:derby://localhost:1527/memory:mydb;create=true<br/>jdbc:derby://localhost:1527/memory:C:\test\mydbmydb;create=true<br/>jdbc:derby://localhost:1527/memory:/test/mydb;create=true|
|Sqlite|`jdbc:sqlite:[subprotocol:][path/][databaseName][?attribute=value]*`|jdbc:sqlite:C:/work/mydatabase.db<br/>jdbc:sqlite:/home/leo/work/mydatabase.db<br/>jdbc:sqlite::memory:<br/>jdbc:sqlite::resource:org/yourdomain/sample.db<br/>jdbc:sqlite::resource:http://www.xerial.org/svn/project/XerialJ/trunk/sqlite-jdbc/src/test/java/org/sqlite/sample.db<br/>jdbc:sqlite::resource:jar:http://www.xerial.org/svn/project/XerialJ/trunk/sqlite-jdbc/src/test/resources/testdb.jar!/sample.db<br/>jdbc:sqlite:db.sqlite?hexkey_mode=sse<br/>jdbc:sqlite::memory:?jdbc.explicit_readonly=true<br/>jdbc:sqlite:sample.db|
|DB2|1、`jdbc:db2:databasename`<br/> 2、`jdbc:db2://[hostname]:[port]/[database_name]:user=[username];password=[password]`|jdbc:db2://localhost:50000/mydb:user=root;password=pass<br/>jdbc:db2:mydb:user=root;password=pass<br/>jdbc:db2://clusterip:50000/mydb|
|Sybase|`jdbc:sybase:Tds:host:port[/database][?connection_property=value;]`|jdbc:sybase:Tds:localhost:5000/testdb?charset=utf-8<br/>jdbc:sybase:Tds:localhost:5000?USER=sa&PASSWORD=secret|
|OceanBase|`jdbc:oceanbase:[hamode:]//host:port/databasename?[username&password]&[opt1=val1&opt2=val2...`]|jdbc:oceanbase://localhost:2881/mydb?user=root@sys&password=pass&pool=false&useBulkStmts=true&rewriteBatchedStatements=false&useServerPrepStmts=true<br/>jdbc:oceanbase://primaryhost:2888,secondaryhost1,secondaryhost2/mydb?user=root@sys&password=pass&pool=false&useBulkStmts=true&rewriteBatchedStatements=false&useServerPrepStmts=true<br/>jdbc:oceanbase:hamode://clusterip:2881/test?user=root@sys&password=pass&pool=false&useBulkStmts=true&rewriteBatchedStatements=false&useServerPrepStmts=true|

Example: Druid + Sqlite can not report metrics before:

General-Root  General-Service  General-Instance

Example: Druid + Sqlite can report metrics now:

image

### CODE REVIEW

File: docs/en/setup/service-agent/java-agent/Supported-list.md [hunk 1]
```diff
@@ -172,6 +172,21 @@ The meter plugin provides the advanced metrics collections, which are not a part
   * [Alibaba Druid](https://github.com/alibaba/druid) 1.x
   * [HikariCP](https://github.com/brettwooldridge/HikariCP) 3.x -> 4.x
   * [C3P0](https://github.com/swaldman/c3p0) 0.9.0 -> 0.10.0
+* Database
```
Comment (wu-sheng) [inline docs/en/setup/service-agent/java-agent/Supported-list.md:175]: I think this should be at lower level. This is a part connection pool.

Comment (CzyerChen) [reply docs/en/setup/service-agent/java-agent/Supported-list.md:175]: OK.

**Comment by wu-sheng** [comment]
@CzyerChen You added RocketMQ plugin test, right? I noticed, it is unstable for various PRs. Could you check the CI logs?

**Review by wu-sheng — APPROVED** [review]

**Comment by CzyerChen** [comment]
> @CzyerChen You added RocketMQ plugin test, right? I noticed, it is unstable for various PRs. Could you check the CI logs?

OK.

**Pull request merged**